index.js

```diff
@@ -32,31 +36,30 @@ SplashScreen.setOptions({
   fade: true,
 });
 
-const defaultTheme = Appearance.getColorScheme();
-
 // https://github.com/facebook/react-native/issues/19410
 if (Platform.OS === "android") {
   require("intl");
   require("intl/locale-data/jsonp/en-US");
 }
 
 const App = () => {
-  const [selectedTheme, setSelectedTheme] = useState(
-    defaultTheme === "dark" ? "dark" : "",
+  Appearance.addChangeListener(({ colorScheme }) => {
+    if (selectedTheme === THEME_SYSTEM_SETTING_VALUE) {
+      // Update theme only if "system" theme is selected
+      updateSelectedTheme(
+        colorScheme === THEME_DARK ? THEME_DARK : THEME_LIGHT,
+      );
+    }
+  });
+
+  const [selectedTheme, updateSelectedTheme] = useState(
```

Comment:
React hook convention favors `set` preceding the variable name being updated via hooks ie `setSelectedTheme`